### PR TITLE
add support for AzeriteUI5's LibActionButton

### PR DESCRIPTION
**Pull request by Goldpaw**
This will add support for the custom UI AzeriteUI5 and its custom version of LibActionButton.

### CODE REVIEW

**Comment by Goldpaw** [comment]
Also note I didn't include LibCustomGlow or LibButtonGlow in the AzeriteUI5 support because neither the UI nor my custom version of LAB supports them. Intentional, not an oversight.

File: components/glow.lua [hunk 1]
```diff
@@ -371,5 +372,38 @@ binder:SetScript("OnEvent", function()
         end
     end
 
+    -- Support for AzeriteUI5's LibActionButton
+    if (LAB_GE) then
+        local buttonUpdateFunc = function(lib, event, self)
```
Comment (ennvina) [inline components/glow.lua:377]: I'm not fond of redefining a function that overrides existing behavior with slight changes. If the previous `buttonUpdateFunc` is updated, this one must be updated as well. I also believe it is confusing if someone else wants to add (yet another) lib action button later on.

I think the same effect can be achieved with this code (not tested yet):
```lua
-- Support for AzeriteUI5's LibActionButton
if (LAB_GE) then
    -- Glow functions crafted for this library
    local libGlowForGE = {
        ShowOverlayGlow = function(button)
            button:SetSpellActivationColor(1, 1, 1);
            button:ShowSpellActivation();
        end,
        HideOverlayGlow = function(button)
            button:HideSpellActivation();
        end
    }
    
    local LAB_GEButtonUpdateFunc = function(event, self)
        buttonUpdateFunc(libGlowForGE, event, self);
    end
end
```

File: components/glow.lua [hunk 2]
```diff
@@ -371,5 +372,38 @@ binder:SetScript("OnEvent", function()
...
+            if (not self.DisableGlow) then
+                self.DisableGlow = function(button)
+                    button:HideSpellActivation();
+                end
+            end
+            SAO:UpdateActionButton(self);
+        end
+
+        local LAB_GEButtonUpdateFunc = function(event, self)
+            buttonUpdateFunc(LAB_GE, event, self);
```
Comment (ennvina) [inline components/glow.lua:402]: `LAB_GE` is passed as argument, but never used in practice. Please read first review comment for suggestion to improve this.

**Review by ennvina — CHANGES_REQUESTED** [review]
Nice work overall. The code could be shorter, easier to read, and reduce technical debt.

File: components/glow.lua (hunk 1 above)
Comment (Goldpaw) [reply components/glow.lua]: I embedded those glow methods into LAB_GE instead, figured that makes even more sense. So there's no longer any need for a modified buttonUpdateFunc or a proxy/dummy library.

File: components/glow.lua (hunk 2 above)
Comment (Goldpaw) [reply components/glow.lua]: As mentioned above, in my latest commit I embedded the glow methods into LAB_GE instead, so now it makes sense to pass LAB_GE as an argument again. 

The latest commit now only checks for the library instance, defines LAB_GEButtonUpdateFunc, and registers it as a callback in LAB_GE if appropriate.

**Comment by Goldpaw** [comment]
Also, not sure I was supposed to press the "resolve conversation" buttons. Consider me a github noob if I did something wrong. My commit from today, October 22nd contains the updated proposed changes in any case.

Sorry for any potential trouble! o.O

File: components/glow.lua (hunk 1 above)
Comment (ennvina) [reply components/glow.lua]: > I embedded those glow methods into LAB_GE instead

What do you mean by that?

Have you modified the code of the AzeriteUI addon itself to define a `ShowOverlayGlow` and `HideOverlayGlow` methods?

**Comment by ennvina** [comment]
> Also, not sure I was supposed to press the "resolve conversation" buttons. Consider me a github noob if I did something wrong. My commit from today, October 22nd contains the updated proposed changes in any case.

@Goldpaw No worries, and thanks for the change.

There's no rule set in stone as to who should press the "resolve conversation" button. Ideally, you (the author) would press it on your side to tell "I think I made necessary changes" and I (the reviewer) would press it on my side to tell "I think this change is correct". But since GitHub does not support a author-approval button and a reviewer-approval button, we have to do it our way.

My go-to approach is:
1. the **reviewer** makes a comment
2. the **author** analyzes the request and either asks for more information, or disagrees, or pushes a change while commenting to explain how the new commit is hopefully better; either way the author does _not_ resolve the conversation
3. the **reviewer** analyzes the reply and either **resolves the conversation** or continues the discussion; the latter sends the author to step 2, rinse and repeat until the conversation is closed by the reviewer

With this approach, step 2 is where you (the author) say "I think it's good now" and step 3 is where I (the reviewer) says "I think it's good now".

Again, there's no rule set in stone, this is what I like to do (after having tested several methods) and should you e.g. work in a company that uses github or gitlab, you might (and you will) see other kinds of pipelines for reviewing.

File: components/glow.lua (hunk 1 above)
Comment (ennvina) [reply components/glow.lua]: I tested it. Although the glowing button is less flashy, it works, as intended by AzeriteUI, I guess.

**Review by ennvina — APPROVED** [review]

**Pull request merged**